### DISCUSSION TITLE
Axiom Support ?

### DISCUSSION THREAD

**Discussion by LemarinelNet** [brefphp/bref#1855] [Ideas]
Hi,

Serverless framework just announced they now support Axiom for logs : 
https://www.serverless.com/framework/docs/guides/observability/axiom

They claim that it is up to 4x cheaper than CloudWatch (which is my biggest source of cost for running a few lambdas...)

How could this be supported also with Bref.sh ? 
And with Bref dashboard ?

**Comment by mnapoli** [comment]
Hey! I haven't looked into how it's working yet, but have you tried it/do you know if it works with Node specifically or if it could work with PHP? Maybe their support can help answer that, not sure if they use a custom Lambda extension or not.

Note that it is possible to disable CloudWatch logs (with or without the Axiom feature), so you can do that today (with Bref) and save on costs. However you need to send your logs somewhere else (datadog, etc.).

(ftr Bref Dashboard is directly tied to CloudWatch)

**Reply by LemarinelNet** [reply]
Hi,
Since it's directly tied to serverless framework, it should work with any supported language I think ? 
Not tried first, but since I mainly use Bref dashboard to view and search in logs, I was checking first if you would be able to add support for it !

I already reduced the retention period for my logs (since the default value was too expensive), but even with a small days window, it cost much more than the lambda execution itself !

Having another option than cloudwatch that would cost 4x less is really interesting !

I'll setup Axiom on my staging env to see if it works out of the box with php and I'll keep you informed here !

Would you be ready to add support for axiom if it's proven as an interesting alternative to cloudwatch ?

PS : I had a look at Axiom documentation, it seems they're lacking a Query API, so it would be rather difficult to add support for it in Bref dashboard :(

PS2 : my bad ! they have rest APIs to query the data, and they even have a JS library :
- https://axiom.co/docs/restapi/introduction
- https://axiom.co/docs/restapi/query

**Reply by mnapoli** [reply]
Sounds interesting, yes!
